File: app/src/main/java/com/example/pizzamaniaapp/app/ui/menu/ItemDetailActivity.java
package com.example.pizzamaniaapp.app.ui.menu;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.bumptech.glide.Glide;
import com.example.pizzamaniaapp.R;

public class ItemDetailActivity extends AppCompatActivity {
    @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_item_detail);

        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("Item details");
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        ImageView iv = findViewById(R.id.ivImage);
        TextView tvTitle = findViewById(R.id.tvTitle);
        TextView tvDesc = findViewById(R.id.tvDesc);
        TextView tvPrice = findViewById(R.id.tvPrice);

        String name = getIntent().getStringExtra("name");
        String desc = getIntent().getStringExtra("desc");
        double price = getIntent().getDoubleExtra("price", 0);
        String imageUrl = getIntent().getStringExtra("imageUrl");

        tvTitle.setText(name != null ? name : "");
        tvDesc.setText(desc != null ? desc : "");
        tvPrice.setText(String.format("Rs. %.2f", price));
        Glide.with(this)
                .load(imageUrl)
                .placeholder(R.mipmap.ic_launcher)
                .into(iv);
    }

    @Override public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) { onBackPressed(); return true; }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/pizzamaniaapp/app/ui/menu/ItemDetailActivity.kt
package com.example.pizzamaniaapp.app.ui.menu

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.example.pizzamaniaapp.R
import com.example.pizzamaniaapp.app.ui.cart.CartActivity

class ItemDetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.row_item_detail) // ✅ your item detail layout

        val backBtn = findViewById<ImageButton>(R.id.btnBack)
        val ivPizza = findViewById<ImageView>(R.id.ivPizza)
        val tvName = findViewById<TextView>(R.id.tvPizzaName)
        val tvDesc = findViewById<TextView>(R.id.tvDescription)
        val tvPrice = findViewById<TextView>(R.id.tvPrice)
        val tvToppings = findViewById<TextView>(R.id.tvToppings)
        val btnAddToCart = findViewById<Button>(R.id.btnAddToCart)

        // Get the data passed from MenuListActivity
        val name = intent.getStringExtra("name")
        val desc = intent.getStringExtra("desc")
        val price = intent.getDoubleExtra("price", 0.0)
        val imgUrl = intent.getStringExtra("imgUrl")
        val toppings = intent.getStringArrayListExtra("toppings_array") // ✅ fixed

        // Bind values
        tvName.text = name
        tvDesc.text = desc
        tvPrice.text = "Rs. %.2f".format(price)
        tvToppings.text =
            if (!toppings.isNullOrEmpty()) "Toppings: ${toppings.joinToString(", ")}"
            else "No toppings"

        Glide.with(this)
            .load(imgUrl)
            .placeholder(R.drawable.placeholder)
            .into(ivPizza)

        // Back button → finish this activity
        backBtn.setOnClickListener { finish() }

        // ✅ Add To Cart → open CartActivity with pizza data
        btnAddToCart.setOnClickListener {
            val intent = Intent(this, CartActivity::class.java).apply {
                putExtra("name", name)
                putExtra("desc", desc)
                putExtra("price", price)
                putExtra("imgUrl", imgUrl)
                putStringArrayListExtra("toppings", toppings)
            }
            startActivity(intent)
        }
    }
}